pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -372,39 +372,78 @@ def required_predicates(self) -> List[Predicate]:
 
         return preds
 
-    def rebase_pass(self, implicit_swaps: bool = False) -> BasePass:
+    @property
+    def _two_qubit_gate_set(self) -> Set[OpType]:
+        """
+        Assumes that only possibly supported two-qubit gates are
+        ZZPhase, ZZMax and TK2.
+
+        :return: Set of two-qubit OpType in gateset.
+        :rtype: Set[OpType]
         """
-        :param implicit_swaps: If true, allows rebasing of Circuit via TK2 gates
-            to use implicit wire swaps in circuit construction if it reduces
-            the total 2qb qate count.
-        :type implicit_swaps: bool
+        supported_2qb_gates: Set[OpType] = set()
+        if OpType.ZZPhase in self._gate_set:
+            supported_2qb_gates.add(OpType.ZZPhase)
+        if OpType.ZZMax in self._gate_set:
+            supported_2qb_gates.add(OpType.ZZMax)
+        if OpType.TK2 in self._gate_set:
+            supported_2qb_gates.add(OpType.TK2)
+        return supported_2qb_gates
+
+    def rebase_pass(self, **kwargs: QuumKwargTypes) -> BasePass:
+        """
+        Supported kwargs:
+        * `implicit_swaps`: Boolean flag, which if true, returns
+            rebasing pass that allows implicit wire swaps.
+            Default False.
+        * `target_2qb_gate`: pytket OpType, if provided, will
+            return a rebasing pass that only allows given
+            two-qubit gate type.
         :return: Compilation pass for rebasing circuits
         :rtype: BasePass
         """
-        return auto_rebase_pass(self._gate_set, implicit_swaps)
+        target_2qb_optype: Union[None, OpType] = kwargs.get("target_2qb_gate")
+        if target_2qb_optype is None:
+            return auto_rebase_pass(
+                self._gate_set, allow_swaps=bool(kwargs.get("implicit_swap", True))
+            )
+        elif target_2qb_optype in self._two_qubit_gate_set:
+            return auto_rebase_pass(
+                self._gate_set - self._two_qubit_gate_set | {target_2qb_optype},
```

Comment:
I don't follow this suggestion, currently the code is set up to allow the user to request targeting any of the two-qubit OpTypes supported by the hardware. The only assumption ("only") on this is that they are in the set {ZZPhase, ZZMax, TK2} based on what we know about the h-series devices. I don't think we need to have special case distinction for ZZMax?

Reply:
I think the current behaviour is wrong though. The submitted circuit is not allowed to contain both ZZPhase and TK2 gates -- otherwise it will be rejected.